mllib/src/main/scala/org/apache/spark/ml/clustering/BisectingKMeans.scala

```diff
@@ -310,4 +317,6 @@ class BisectingKMeansSummary private[clustering] (
     predictionCol: String,
     featuresCol: String,
     k: Int,
-    numIter: Int) extends ClusteringSummary(predictions, predictionCol, featuresCol, k, numIter)
+    numIter: Int,
+    @Since("2.4.0") val trainingCost: Double)
```

Comment:
2.4.0? or 3.0.0?

Reply:
this PR targets to 2.4, see more context at https://github.com/apache/spark/pull/22756

Reply:
oh wait. If the final goal is to have a consistent ML API in 3.0, do we have to put this new API in 2.4?

Reply:
I wouldn't consider this as mandatory. I think what is mandatory to target for 2.4 is to deprecate the `computeCost` method. But I think it is a nice to have, since it is a non-deprecated way users have to access this information. In the PR related to `KMeans`, there was quite a discussion about it and it was considered to be part of the deprecation change.

Reply:
SGTM

Reply:
Ok. I see. Then looks like it is nice to have this in 2.4.

Reply:
Since #22756 is reverted, are we going to change the `Since` version for this, too?